### PR TITLE
Add auto Runtime.stop()

### PR DESCRIPTION
**Pull request by mgkwill**

### CODE REVIEW

File: src/lava/magma/core/process/process.py [hunk 1]
```diff
@@ -249,6 +249,12 @@ def __init__(self, **kwargs):
         # Current runtime environment
         self._runtime: ty.Optional[Runtime] = None
 
+    def __del__(self):
+        """On destructor call automatically
```
Comment (awintel) [inline src/lava/magma/core/process/process.py:253]: Alternative docstring suggestion: "On destruction, terminate Runtime automatically to free compute resources."

File: src/lava/magma/runtime/runtime.py [hunk 2]
```diff
@@ -64,6 +64,13 @@ def __init__(self,
         self.service_to_runtime_data: ty.Iterable[CspRecvPort] = []
         self.runtime_to_service_data: ty.Iterable[CspSendPort] = []
 
+    def __del__(self):
+        """On destructor call automatically
```
Comment (awintel) [inline src/lava/magma/runtime/runtime.py:68]: Alternative docstring suggestion: "On destruction, terminate Runtime automatically to free compute resources."

**Review by awintel — APPROVED** [review]
In principle all good, just some minor docstring suggestions.
Thanks!

**Review by mgkwill — COMMENTED** [review]
LGTM.
@jlakness-intel @PhilippPlank care to review?

**Review by PhilippPlank — APPROVED** [review]
Looks fine for me.

**Pull request merged**